tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -577,6 +606,12 @@ TEST_F(InputProcessorFixture, parse_bad_utf_8_json_3)
           "\"starting_vertex_position\":\"UpperLeftCorner\","
           "\"vertex_entry_direction\":\"Counterclockwise\""
         "}"
+      "},"
+      "\"Version\":{"
+        "\"\":{"
+          "\"idf_order\":0,"
```

Comment:
You shouldn't need this anymore and if you do for the test to pass then I consider that a bug.

Reply:
https://github.com/NREL/EnergyPlus/blob/b86e836101bd67a64d95b8b889ff508fa2f8251f/tst/EnergyPlus/unit/InputProcessor.unit.cc#L610-L615

Reply:
`process_idf` will add the Version object, so I do need to add it to the `expected` much like GlobalGeometryRules. Why is that a bug?

Reply:
@jmarrec I mean specifically `"\"idf_order\":0,"` isn't needed. I know it was still in the GlobalGeometryRules and Building objects that are added automatically, but I forgot to remove those once those fields were no longer necessary. If `process_idf` requires `"idf_order":0` still after you remove it from that function, that is what I consider a bug.

Reply:
Ah ok thanks for the clarification. I copied the idf_order over without second thought here

Reply:
@mbadams5 I removed the `idf_order` for Building, GlobalGeometry, and Version in this test, and it now fails:

```
[ RUN      ] InputProcessorFixture.parse_bad_utf_8_json_3
/home/julien/Software/Others/EnergyPlus/tst/EnergyPlus/unit/InputProcessor.unit.cc:614: Failure
      Expected: expected
      Which is: "{\"Building\":{\"\xEF\xBF\xBD\xEF\xBF\xBD\xEF\xBF\xBD\":{\"idf_max_extensible_fields\":0,\"idf_max_fields\":8,}},\"GlobalGeometryRules\":{\"\":{\"coordinate_system\":\"Relative\",\"daylighting_reference_point_coordinate_system\":\"Relative\",\"rectangular_surface_coordinate_system\":\"Relative\",\"starting_vertex_position\":\"UpperLeftCorner\",\"vertex_entry_direction\":\"Counterclockwise\"}},\"Version\":{\"\":{\"version_identifier\":\"9.4\"}}}"
To be equal to: input_file
      Which is: "{\"Building\":{\"\xEF\xBF\xBD\xEF\xBF\xBD\xEF\xBF\xBD\":{\"idf_max_extensible_fields\":0,\"idf_max_fields\":8,\"idf_order\":1}},\"GlobalGeometryRules\":{\"\":{\"coordinate_system\":\"Relative\",\"daylighting_reference_point_coordinate_system\":\"Relative\",\"idf_order\":0,\"rectangular_surface_coordinate_system\":\"Relative\",\"starting_vertex_position\":\"UpperLeftCorner\",\"vertex_entry_direction\":\"Counterclockwise\"}},\"Version\":{\"\":{\"idf_order\":0,\"version_identifier\":\"9.4\"}}}"
```